pkg/api/configsync/v1beta1/helmconfig.go

```diff
@@ -93,3 +117,19 @@ type HelmBase struct {
 	// +optional
 	SecretRef *SecretReference `json:"secretRef,omitempty"`
 }
+
+// ValuesFrom holds references to objects in the cluster that represent
+// values to use instead of default values that accompany the chart.
+type ValuesFrom struct {
+	// kind represents the Object kind. Must be `ConfigMap`. Required.
+	// +kubebuilder:validation:Enum=ConfigMap
+	Kind string `json:"kind,omitempty"`
+
+	// name represents the Object name. Required.
+	Name string `json:"name,omitempty"`
+
+	// key represents the object data key to read the value from.
+	// +kubebuilder:default:=values.yaml
+	// +optional
+	Key string `json:"key,omitempty"`
```

Comment:
The name "Key" seems specific to the Kind ConfigMap.
If this were any other resource type, Key wouldn't make any sense.
Should it be "Field" instead?
Should it support JSONPath to support other resource types?

Reply:
If we only support ConfigMaps, do we need this complexity of specifying Kind and Key?